org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/ParameterizedRecordComponentBinding.java

```diff
@@ -0,0 +1,68 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * This is an implementation of an early-draft specification developed under the Java
+ * Community Process (JCP) and is made available for testing and evaluation purposes
+ * only. The code is not compatible with any specification of the JCP.
+ *
+ * Contributors:m
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.internal.compiler.lookup;
+
+import org.eclipse.jdt.internal.compiler.classfmt.ClassFileConstants;
+import org.eclipse.jdt.internal.compiler.impl.Constant;
+
+/**
+ * Binding denoting a record component after type substitution got performed.
+ * On parameterized type bindings, all record components got substituted, regardless whether
+ * their signature did involve generics or not, so as to get the proper declaringClass for
+ * these record components.
+ */
+public class ParameterizedRecordComponentBinding extends RecordComponentBinding {
+
+    public RecordComponentBinding originalRecordComponent;
+
+public ParameterizedRecordComponentBinding(ParameterizedTypeBinding parameterizedDeclaringClass, RecordComponentBinding originalRecordComponent) {
+	super (originalRecordComponent.name,
+            (originalRecordComponent.modifiers & ClassFileConstants.AccEnum) != 0
```

Comment:
copied from `ParameterizedFieldBinding`? A field can be an enum constant, but can a record component?